infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/SinkWriter.java

```diff
@@ -80,14 +83,20 @@ List<SMREntry> filterRegistryTableEntries(List<SMREntry> smrEntries) {
     }
 
     /**
-     * Check if a stream id belongs to list of replicated streams to drop in LogReplicationConfig. If so, its entries
+     * Check if a stream id belongs to list of replicated streams to drop in LogReplicationFullTableConfig. If so, its entries
      * should be ignored by SnapshotWriter and LogEntryWriter.
      *
      * @param streamId ID of the stream whose entries are being applied by LR
      * @return True if the entries should be ignored.
      */
     boolean ignoreEntriesForStream(UUID streamId) {
-        return replicationContext.getConfig().getStreamsToDrop().contains(streamId);
+        if (session.getSubscriber().getModel().equals(LogReplication.ReplicationModel.FULL_TABLE)) {
+            return ((LogReplicationFullTableConfig) replicationContext.getConfig(session))
+                    .getStreamsToDrop().contains(streamId);
+        } else {
+            log.warn("Unexpected path for the sink writer of current replication session {}", session);
```

Comment:
this is not an unexpected case.  It is valid that this check gets invoked for other replication models as well.  Can you change the log message accordingly?  Also, please add a TODO as we must handle the streams to drop for other models also.

Reply:
Discussed offline and added a TODO here